codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -386,13 +383,57 @@ private void generateOperationHttpBindingSerializer(
     }
 
     private Symbol getRestEncoderSymbol() {
-        return SymbolUtils.createPointableSymbolBuilder("Encoder", GoDependency.AWS_REST_PROTOCOL).build();
+        return SymbolUtils.createPointableSymbolBuilder("Encoder", GoDependency.SMITHY_HTTP_BINDING).build();
     }
 
-    private String generateHttpBindingSetter(
+    private void generateHttpBindingTimestampSerializer(
             Model model,
+            GoWriter writer,
             MemberShape memberShape,
-            String operand
+            HttpBinding.Location location,
+            String operand,
+            BiConsumer<GoWriter, String> locationEncoder
+    ) {
+        writer.addUseImports(GoDependency.SMITHY_TIME);
+
+        TimestampFormatTrait.Format format = memberShape.getMemberTrait(model, TimestampFormatTrait.class)
+                .map(TimestampFormatTrait::getFormat)
+                .orElseGet(() -> {
+                    switch (location) {
+                        case LABEL:
+                        case QUERY:
+                            return Format.DATE_TIME;
+                        case HEADER:
+                        case PREFIX_HEADERS:
+                            return Format.HTTP_DATE;
+                        default:
+                            throw new CodegenException("Unexpected HttpBinding location for timestamp");
+                    }
+                });
```

Comment:
Assuming these are consistent between REST-JSON and REST-XML but wanted to check and make sure.

Reply:
These are the default values for http binding traits for both protocols if TimestampFormatTrait is not present.

file:///Volumes/Brazil/workspace/sdk/smithy/docs/build/html/spec/aws/aws-restjson1-protocol.html#supported-traits